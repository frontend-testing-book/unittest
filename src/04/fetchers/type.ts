export type HttpError = {
  err: { message: string }
}

export type Profile = {
  id: string
  name?: string
  age?: number
  email: string
}

export type Article = {
  id: string
  createdAt: string
  tags: string[]
  title: string
  body: string
}

export type Articles = {
  articles: Article[]
}

export type ArticleInput = {
  tags: string[]
  title: string
  body: string
}

export type PartialArticleInput = {
  tags?: string[]
  title?: string
  body?: string
}
