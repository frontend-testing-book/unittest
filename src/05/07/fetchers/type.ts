export type HttpError = {
  err: { message: string }
}

export type Result = {
  result: string
}
