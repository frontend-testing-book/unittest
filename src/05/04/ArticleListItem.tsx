export type ItemProps = {
  id: string;
  title: string;
  body: string;
};

async function postLog() {
  // ログを送信する処理
  return;
}

export const ArticleListItem = ({ id, title, body }: ItemProps) => {
  return (
    <li>
      <h3>{title}</h3>
      <p>{body}</p>
      <a
        style={{
          color: "blue",
          cursor: "pointer",
          textDecoration: "underline",
        }}
        onClick={async (event) => {
          const href = `/articles/${id}`;
          event.preventDefault();
          await postLog();
          window.location.assign(href);
        }}
      >
        もっと見る
      </a>
    </li>
  );
};
