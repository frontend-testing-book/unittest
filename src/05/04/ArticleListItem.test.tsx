import { render, screen } from '@testing-library/react'

import type { ItemProps } from './ArticleListItem'
import { ArticleListItem } from './ArticleListItem'

const item: ItemProps = {
  id: 'howto-testing-with-typescript',
  title: 'TypeScript を使ったテストの書き方',
  body: 'テストを書く時、TypeScript を使うことで、テストの保守性が向上します…',
}

test('ID に紐づいたリンクが表示される', () => {
  render(<ArticleListItem {...item} />)
  expect(screen.getByRole('link', { name: 'もっと見る' })).toHaveAttribute(
    'href',
    '/articles/howto-testing-with-typescript',
  )
})

test('Snapshot: 一覧要素が表示される', () => {
  const { container } = render(<ArticleListItem {...item} />)
  expect(container).toMatchSnapshot()
})
