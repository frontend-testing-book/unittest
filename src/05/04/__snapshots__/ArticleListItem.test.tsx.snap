// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: 一覧要素が表示される 1`] = `
<div>
  <li>
    <h3>
      TypeScript を使ったテストの書き方
    </h3>
    <p>
      テストを書く時、TypeScript を使うことで、テストの保守性が向上します…
    </p>
    <a
      style="color: blue; cursor: pointer; text-decoration: underline;"
    >
      もっと見る
    </a>
  </li>
</div>
`;
